### PR TITLE
Unique message id mapper - Replaced by #482

### PR DESCRIPTION
**Pull request by aduprat**
Additional commits to #463 

Each commits will be squashed with initial commits: API, implementation & tests.
Each bunch of commits comes with three commits: API, implementation & tests.

- find returns MailboxMessage instead of Message
MAILBOX-277 API: find should return mailbox messages 2e2ccba
MAILBOX-277 Cassandra: Find should return MailboxMessage 61b79e3
fixup! MAILBOX-277 MessageIdMapper test a95469c

- add delete on given mailboxes method
MAILBOX-277 API: Provide per mailboxes deletion with MessageIdMappers e7f643f
MAILBOX-277 Cassandra: Provide per mailbox deletion with MessageIdMap b15c2f0
fixup! MAILBOX-277 MessageIdMapper test 687aeae

- add setFlags method
MAILBOX-277 API: Interfaces for changing flags ff7bfcc
MAILBOX-277 Add methods with mailboxId parameter in ModSeqProvider fa24707
MAILBOX-277 Cassandra: Interfaces for changing flags 3f29e42
fixup! MAILBOX-277 MessageIdMapper test f86600b

- Ease testing
MAILBOX-277 Add toString in UpdatedFlags class 6a60949

### CODE REVIEW

**Comment by aduprat** [comment]
Rebased on master, push -f -> then GitHub breaks the commit order...

**Comment by aduprat** [comment]
test this please

**Comment by aduprat** [comment]
test this please

File: mailbox/store/src/test/java/org/apache/james/mailbox/store/mail/model/MessageIdMapperTest.java [hunk 1]
```diff
@@ -240,6 +240,66 @@ public void deleteShouldDeleteMessageIndicesWhenStoredTwoTimesInTheSameMailbox()
         assertThat(mailboxes).isEmpty();
     }
 
+    @ContractTest
```
Comment (chibenwa) [inline mailbox/store/src/test/java/org/apache/james/mailbox/store/mail/model/MessageIdMapperTest.java]: Aren't we supposed to delete the message when there's no more references to it ?

File: mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/CassandraMailboxSessionMapperFactory.java [hunk 2]
```diff
@@ -87,7 +87,7 @@ public CassandraMessageMapper createMessageMapper(MailboxSession mailboxSession)
     @Override
```
Comment (chibenwa) [inline mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/CassandraMailboxSessionMapperFactory.java]: Aren't we supposed to have some tests on the UpdatedFlags returned here ?

**Comment by aduprat** [comment]
This review #482 has the good commit order.

**Comment by aduprat** [comment]
Please review this one (rebased): #482

**Pull request closed without merging**